src/batch/src/executor/source.rs

```diff
@@ -12,19 +12,193 @@
+use risingwave_common::types::DataType;
+use risingwave_connector::source::{ConnectorProperties, SplitImpl, SplitMetaData};
+use risingwave_pb::batch_plan::plan_node::NodeBody;
+use risingwave_pb::catalog::source_info::SourceInfo as ProstSourceInfo;
+use risingwave_pb::plan_common::RowFormatType;
+use risingwave_source::connector_source::{ConnectorSource, SourceContext};
+use risingwave_source::monitor::SourceMetrics;
+use risingwave_source::{SourceColumnDesc, SourceFormat, SourceParserImpl};
+
+use super::Executor;
 use crate::executor::{BoxedExecutor, BoxedExecutorBuilder, ExecutorBuilder};
 use crate::task::BatchTaskContext;
 
-pub struct SourceExecutor {}
+pub struct SourceExecutor {
+    connector_source: ConnectorSource,
+
+    // used to create reader
+    column_ids: Vec<ColumnId>,
+    metrics: Arc<SourceMetrics>,
+    source_id: TableId,
+    split: SplitImpl,
+
+    schema: Schema,
+    identity: String,
+}
 
 #[async_trait::async_trait]
 impl BoxedExecutorBuilder for SourceExecutor {
     async fn new_boxed_executor<C: BatchTaskContext>(
-        _source: &ExecutorBuilder<'_, C>,
-        _inputs: Vec<BoxedExecutor>,
+        source: &ExecutorBuilder<'_, C>,
+        inputs: Vec<BoxedExecutor>,
     ) -> Result<BoxedExecutor> {
-        todo!("Can't support SourceExecutor now!")
+        ensure!(inputs.is_empty(), "Source should not have input executor!");
+        let source_node = try_match_expand!(
+            source.plan_node().get_node_body().unwrap(),
+            NodeBody::Source
+        )?;
+        match source_node
+            .properties
+            .get("connector")
+            .expect("source must have connector property")
+            .to_lowercase()
+            .as_str()
+        {
+            "kafka" | "kinesis" | "pulsar" => (),
+            other => {
+                return Err(ErrorCode::BatchError(
+                    anyhow!("Unsupported batch source type: {}", other).into(),
+                )
+                .into());
+            }
+        }
+
+        // prepare connector source
+        let source_props: HashMap<String, String> =
+            HashMap::from_iter(source_node.properties.clone().into_iter());
+        let config = ConnectorProperties::extract(source_props)
+            .map_err(|e| RwError::from(ConnectorError(e.into())))?;
+
+        let info = try_match_expand!(
+            &source_node.get_info()?.get_source_info()?,
+            ProstSourceInfo::StreamSource
+        )
+        .unwrap();
+        let format = match info.get_row_format()? {
+            RowFormatType::Json => SourceFormat::Json,
+            RowFormatType::Protobuf => SourceFormat::Protobuf,
+            RowFormatType::DebeziumJson => SourceFormat::DebeziumJson,
+            RowFormatType::Avro => SourceFormat::Avro,
+            RowFormatType::Maxwell => SourceFormat::Maxwell,
+            RowFormatType::CanalJson => SourceFormat::CanalJson,
+            RowFormatType::RowUnspecified => unreachable!(),
```

Comment:
We should return error here rather panic.

Reply:
https://github.com/risingwavelabs/risingwave/blob/ae398b3b550a986da0cb1a33fbe0629f90ae9a3f/src/source/src/manager.rs#L260

Do I aslo need to modify here?

Reply:
I see, this is protobuf default enum value. Seems we can only panic here.

Reply:
if row format is not provided, it will raise an error in the frontend and it won't go that far